### ISSUE TITLE
Ensure that `nav ul` styles are not bad for accessibility

### ISSUE THREAD

**Issue by aarongarciah** [mozdevs/cssremedy#15]
After reading this article by @scottaohara https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html I want to open a discussion about this particular rule (`list-style: none`) https://github.com/mozdevs/cssremedy/blob/8108c97ce590aed9e315fce4b72824f3b09faffc/remedy.css#L68

There are evidences that it is problematic using VoiceOver with Safari (macOS and iOS), it removes the list semantics.

As @scottaohara points in the article, using one of these rules maybe is better to avoid removing list semantics:

```css
list-style: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg'/%3E");

// or

list-style: url(#!);
```

It's a bit hacky, maybe it's worth it.

PS: thanks for this project 👌

**Comment by scottaohara** [comment]
Hey @aarongarciah, thanks for reading / the ping.

It's worth noting that I found some instances of visual quirks with IE11 and Edge per the method you mentioned, as I referenced in that section of the article.

The following snippet works much more consistently as a CSS method to re-introduce list semantics for unstyled lists:

```css
.list li {
  list-style-type: none; /* remove bullets */
}
 
.list li:before {
  content: "\200B"; /* add zero-width space */
  position: absolute; /* ensures it doesn't cause a gap */
}
```

It's also worth noting reasoning was given for Webkit's behavior here, and some of the proposed revisions (e.g. Webkit continuing to remove semantics of lists within navs, but not doing so outside of a nav), took place in [this twitter thread](https://twitter.com/cookiecrook/status/1084142936583892992).

As the very least, if no change were made to the CSS here, that a comment would be added to note that such a rule would remove semantics in Webkit based browsers, and options to re-add those semantics if the developer choose to do so.

**Comment by jensimmons** [comment]
>This was a purposeful change due to rampant "list"-itis by web developers. In the same way WebKit differentiates layout tables from data tables, it now differentiates layout lists from data lists. VoiceOver users always used to file bugs that there were too many "start of list" "end of list" announcements on the Web. Customers seems much happier in the 3 years since this change went in.
James Craig, [explaining why](https://bugs.webkit.org/show_bug.cgi?id=170179#c1) VoiceOver doesn't announce lists with `list-style: none;`. 

This reminds me of the "to-do" in the back of my mind to research.... should menus really be lists in the first place? What's wrong with 
```
<nav>
    <a href="example.com">Home</a>
    <a href="example.com">Home</a>
    <a href="example.com">Home</a>
</nav>
```
or any other non-list markup that might be used. I've been wanting to talk to screen-reader experts to know how this impacts the a11y screen-reader experience. 

BTW, `li { list-style-type: none; }` isn't the only way to remove the bullets. It's the one we've been teaching each other for years, but I've also wondered whether or not it's best.

A list has a bullet because it is given a marker box when `li { display: list-item;}` is applied by the UA stylesheet. The easiest way to remove the marker — and the whole marker box — is to change the formatting context of the list item with `li {display: block};`, `li {display: inline; }`, or `li { display: inline-block; }`. (Which does not remove the margin or padding that's on the `ul` to make space for the marker box, so that has to be done separately.)

Anyway, back to the main issue. If the makers of VoiceOver removed the list-ness of such lists because it was annoying to users, do we want to hack around their decision to put it back in? I'm not sure we should.

**Comment by aarongarciah** [comment]
> If the makers of VoiceOver removed the list-ness of such lists because it was annoying to users, do we want to hack around their decision and force it back in? I'm not sure we should.

Very good point @jensimmons. As you say, it would be awesome to know the opinion of screen readers experts about this topic. 

Thanks.

**Comment by gerardkcohen** [comment]
I'd start off by saying that I don't think that CSS should ever interfere with HTML semantics. Understanding that VO (VoiceOver) users may have been annoyed by the proliferation of List usage and announcements, I think it would have been better for this to land as a user option in VO rather than overwriting the semantics of the list.

That being said, the context for my decision to come up with a solution had to do with the fact that it is part of a larger internal framework, where number of items can never be accurately predicted or assumed. Navigation needs to support a few items (breadcrumbs, simple footer links) up to mega menus that are nested and have sub-items. It's in these larger lists that I felt it was important to provide users with at least a count of items included before deciding to navigate the set with an understanding that AT (Assistive Technology) users can/ will cancel any extraneous announcements they are not interested in. When information can be useful to some, we usually choose to be a little more verbose vs not providing that info at all.

My recommendation internally was to add this as a specific utility class instead of applying it across the board. This way, the author can have full control over markup choices, while having the AT respect the intention. Again, this is for enterprise-class applications and may not apply to everyone. However, if we look at sites like [Simply Accessible](https://simplyaccessible.com/), [The Paciello Group](https://www.paciellogroup.com/), [Deque](https://www.deque.com/), [Knowbility](https://knowbility.org/), and [Level Access](https://www.levelaccess.com/), we can take some cues that something like `nav` > `ul` is not so bad.

cc: @jensimmons  Apologies for the late response.

**Comment by reubenlillie** [comment]
> This reminds me of the "to-do" in the back of my mind to research.... should menus really be lists in the first place? What's wrong with

```html 
<nav>
    <a href="example.com/">Home</a>
    <a href="example.com/about">About</a>
    <a href="example.com/contact">Contact</a>
</nav>
```

I'd also like to hear from a11y experts.

I'm not sure there's anything implictly "wrong" with anchors-as-direct-children (except, maybe, that `<a>` are inline elements). In fact, semanticly, I think it's a default practice worth considering. The primary exception I can think of would be `<nav>` that are list-like (e.g., a table of contents for an HTML version of a book). In that case, I can imagine screen reader users might appreciate an announcement even if `list-style: none;` is visually applied.

**Comment by Nuc1eoN** [comment]
Chris Coyier published a good article on this topic on [css-tricks.com](https://css-tricks.com/wrapup-of-navigation-in-lists/)

In fact there have been several such articles over time (all linked in the above mentioned post) and it seem as of yet, there is no conclusive answer.

Fun fact, the website css-tricks.com itself implemented listless navigation some time back, but then chose to revert back to list style navigation over time.

**Comment by scottaohara** [comment]
lists provide a sense to how many items are within a navigation.  navs without lists provide no such information.

even in the most recent back and forth on the topic it was noted that maybe the heuristics should be [updated specifically within navigations](https://twitter.com/rogerjohansson/status/1084142422098100226?s=20)

**Comment by mirisuzanne** [comment]
closed by #38

**Comment by mirisuzanne** [comment]
(I'll also be updating it slightly in #51 to add @scottaohara's `position: absolute` suggestion)
